rfcs/010-per-document-access-control.md

```diff
@@ -0,0 +1,418 @@
+
+*developer*: creator of an application built on top of CouchDB
+
+## Requirements Language
+
+[NOTE]: # ( Do not alter the section below. Follow its instructions. )
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in
+[RFC 2119](https://www.rfc-editor.org/rfc/rfc2119.txt).
+
+---
+
+# Detailed Description
+
+You will be able to create databases with the “access” feature enabled
+via an option passed at database creation time. If you create a
+database without that option, it works like any database in CouchDB
+today.
+
+This is how you create an access-enabled database:
+
+```
+PUT /database?access=true
+```
+
+This option can be set only at database creation time, it can’t be
+turned off and on while the database exists.
+
+An access-enabled database behaves like this:
+
+* only admin users can read or write to the database (as per 3.x
+  defaults)
+
+* admins can grant individual users and groups access to a database
+  using the database’s `_security` object. A special new role `_users`
+  can be used to say “all users defined in the `_users` database”.
+
+* documents created without an `_access` field are accessible to
+  db-admins only
+
+   * this allows existing databases to be replicated into an
+     access-enabled database, but granting access of individual docs to
+     specific users needs to be an explicit step handled by developers.
+
+* documents created with an `_access` field are only accessible by
+  admins and the user named inside `_access`.
+
+   * `_access: ["shirley"]`
+
+      * later iterations of this could allow for `["shirley"]` being
+        shorthand for `[{"read": "shirley", "write": "shirley"}]` for
+        more fine-grained access control, but that is out of scope for
+        this RFC.
+
+   * users can only create documents with their own username in
+     `_access`.
+
+   * admins can add any users to `_access`
+
+   * documents can only be owned by one user at any one point.
+
+      * in a 2.0 > X < 4.0 cluster, two different users could create
+        the same document with a different _access definition
+        concurrently and both get successful write responses back. As
+        with _users documents in conflict, if a document has a conflict
+        with separate _access entries, it becomes admin-only by
+        default. This case needs to be handled by an applications
+        _conflict handler.
+
+* document _ids are shared across all users. So only the first user who
+  creates the doc `_id: config` gets it. Applications need to ensure to
+  work around this and potentially prefix docs with the username before
+  writing/replicating them in.
+
+* _security members are allowed to write design docs, but they have to
+  have an `_access` field and those design docs with an `_access` field
+  are ignored on the server side. Db-admin ddocs get indexes built as
```

Comment:
I think it might be clearer if you write this as: 
design doc's with  an `_access` field will be ignored in an access database

Reply:
yeah I must have missed this one, I have it that way further down